src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Automation/VsBuildManager.cs

```diff
@@ -0,0 +1,180 @@
+        {
+            AddEventSource(this as IEventSource<_dispBuildManagerEvents>);
+            _vsProject = vsProject;
+            _unconfiguredProjectServices = unconfiguredProjectServices;
+        }
+
+        #region _dispBuildManagerEvents_Event Members
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public event _dispBuildManagerEvents_DesignTimeOutputDeletedEventHandler DesignTimeOutputDeleted;
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public event _dispBuildManagerEvents_DesignTimeOutputDirtyEventHandler DesignTimeOutputDirty;
+
+        #endregion
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public virtual EnvDTE.Project ContainingProject
+        {
+            get { return _vsProject.Project; }
+        }
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public virtual EnvDTE.DTE DTE
+        {
+            get { return _vsProject.DTE; }
+        }
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        public virtual object DesignTimeOutputMonikers
+        {
+            get
+            {
+                var project = _unconfiguredProjectServices.ActiveConfiguredProject;
+                var ruleSource = project.Services.ProjectSubscription.ProjectRuleSource;
+                var update = _unconfiguredProjectServices.ThreadingService.ExecuteSynchronously(() => ruleSource.GetLatestVersionAsync(project, new string[] { Compile.SchemaName }));
+                var snapshot = update[Compile.SchemaName];
+
+                var monikers = new List<string>();
+                foreach (var item in snapshot.Items.Values)
+                {
+                    bool isLink = GetBooleanPropertyValue(item, Compile.LinkProperty);
+                    bool designTime = GetBooleanPropertyValue(item, Compile.DesignTimeProperty);
+
+                    if (!isLink && designTime)
+                    {
+                        if (item.TryGetValue(Compile.FullPathProperty, out string path))
+                        {
+                            monikers.Add(path);
+                        }
+                    }
+                }
+                
+                return monikers.ToArray();
+
+                bool GetBooleanPropertyValue(System.Collections.Immutable.IImmutableDictionary<string, string> item, string propertyName)
+                {
+                    return item.TryGetValue(propertyName, out string value) && StringComparers.PropertyValues.Equals(value, "true");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Undocumented.
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1065:DoNotRaiseExceptionsInUnexpectedLocations")]
+        public virtual object Parent
+        {
+            get { throw new NotImplementedException(); }
```

Comment:
Why virtual, and what does this return for legacy?

Reply:
This returns VSProject in legacy.

Reply:
The virtual is just because I copied this file direct from CPS. Will clean it up.

The only thing worth looking at in this file at the moment is the DesignTimeOutputMonikers property, but in legacy this returns the project, so presumably _vsProject.

Reply:
This needs to be "our" VSProject not CPS's version.